models/particle_net.py

```diff
@@ -0,0 +1,160 @@
+from tkinter import W
+from omegaconf import OmegaConf, DictConfig
+import tensorflow as tf
+from tensorflow.keras.layers import BatchNormalization, Dense, Dropout, Conv2D, Activation
+
+
+class ParticleNet(tf.keras.Model):
+    def __init__(self, feature_name_to_idx, encoder_cfg, decoder_cfg):
+        super(ParticleNet, self).__init__()
+
+        self.feature_name_to_idx = feature_name_to_idx
+        self.num_classes = decoder_cfg['n_outputs']
+        self.conv_params = [(layer_setup[0], tuple(layer_setup[1]),) for layer_setup in encoder_cfg['conv_params']]
+        self.conv_pooling = encoder_cfg['conv_pooling']
+        self.fc_params = [(layer_setup, decoder_cfg['dropout_rate'],) for layer_setup in decoder_cfg['dense_params']]
+        self.num_points = encoder_cfg['seq_cutoff_len']
+        self.masking = encoder_cfg['masking']
+
+        # Initialising layers
+        self.batch_norm = BatchNormalization()
+        
+        self.edge_conv_layers = []
+        for layer_idx, layer_param in enumerate(self.conv_params):
+            K, channels = layer_param
+            self.edge_conv_layers.append(
+                EdgeConv(self.num_points, K, channels, with_bn=True, activation='relu', pooling=self.conv_pooling, name=f'{self.name}_EdgeConv_{layer_idx}')
+            )
+
+        if self.fc_params is not None:
+            self.decoder_layers = tf.keras.Sequential()
+
+            for layer_idx, layer_param in enumerate(self.fc_params):
+                units, dropout_rate = layer_param
+
+                self.decoder_layers.add(Dense(units, activation='relu'))                
+                if dropout_rate is not None and dropout_rate > 0:
+                    self.decoder_layers.add(Dropout(dropout_rate))
+
+            self.decoder_layers.add(Dense(self.num_classes, activation='softmax'))
```

Comment:
tabulate it out of the `if` statement, so that this layer is always added:
```suggestion
        self.decoder_layers.add(Dense(self.num_classes, activation='softmax'))
```